### PR TITLE
[PRM-2354] Moved header outside outside radios and date pages

### PR DESCRIPTION
**Pull request by JBlacklaws**

### CODE REVIEW

**Comment by JBlacklaws** [comment]
Desktop: 
Screenshot 2023-03-06 at 10 58 35

Mobile:
Screenshot 2023-03-06 at 11 08 14

HTML:
Screenshot 2023-03-06 at 10 59 34

File: app/views/CancelVATRegistrationPage.scala.html [hunk 1]
```diff
@@ -27,12 +28,14 @@
     formHasErrors = form.errors.nonEmpty,
     formErrors = form.errors,
     optPageMode = Some(pageMode)) {
+    @pageHeader(messages("cancelVATRegistration.headingAndTitle"))
```
Comment (LewisMitchell) [inline app/views/CancelVATRegistrationPage.scala.html:31]: The caption should be a h2 outside of the fieldset

File: app/views/CancelVATRegistrationPage.scala.html [hunk 2]
```diff
@@ -27,12 +28,14 @@
...
     @formHelper(action = postAction) {
         @radios(
             form = form,
-            legendContent = messages("cancelVATRegistration.headingAndTitle"),
+            legendContent = "",
```
Comment (LewisMitchell) [inline app/views/CancelVATRegistrationPage.scala.html:35]: The legend should still contain the page heading

File: app/views/CancelVATRegistrationPage.scala.html [hunk 3]
```diff
@@ -18,7 +18,8 @@
 bullets: components.bullets,
 formHelper: FormWithCSRF,
 button: components.button,
-radios: components.radios)
+radios: components.radios,
+pageHeader: components.pageHeader)
```
Comment (LewisMitchell) [inline app/views/CancelVATRegistrationPage.scala.html:23]: The latest design can be found here: https://design.tax.service.gov.uk/hmrc-design-patterns/page-heading/#options-example-legends-as-page-headings-with-subheading

```
<h2>caption here</h2>
<fieldset>
<legend>
<h1>page heading here</h1>
</legened>
</fieldset>
```

I believe this should be applied on several pages that make use of a fieldset (or radios in this case)

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Comment by LewisMitchell** [comment]
Could you rebase again please?

File: app/views/CancelVATRegistrationPage.scala.html [hunk 4]
```diff
@@ -27,12 +28,15 @@
     formHasErrors = form.errors.nonEmpty,
     formErrors = form.errors,
     optPageMode = Some(pageMode)) {
+    @pageHeader(messages("cancelVATRegistration.headingAndTitle"), showPageHeader = false)
```
Comment (LewisMitchell) [inline app/views/CancelVATRegistrationPage.scala.html:31]: Maybe `showPageHeader` isn't needed? Could we change the first parameter of the page header to be Option[String]?

File: app/views/components/pageHeader.scala.html [hunk 5]
```diff
@@ -36,6 +36,8 @@ <h2 class="govuk-caption-l hmrc-caption-l govuk-!-margin-top-1" id="page-caption
   }
 </h2>
 
-<h1 class="govuk-heading-l govuk-!-margin-bottom-4" id="page-heading">
-  @pageHeader
-</h1>
+@if(showPageHeader) {
```
Comment (LewisMitchell) [inline app/views/components/pageHeader.scala.html:39]: Further to the above, if we change pageHeader to Option[String] we could get rid of the showPageHeader param? What do you think?

Comment (LewisMitchell) [reply app/views/components/pageHeader.scala.html:39]: This check would then become pageHeader.isDefined

File: app/views/CancelVATRegistrationPage.scala.html [hunk 6]
```diff
@@ -27,12 +28,15 @@
     formHasErrors = form.errors.nonEmpty,
```
Comment (LewisMitchell) [inline app/views/CancelVATRegistrationPage.scala.html:28]: I believe date inputs also use fieldsets so we made need to change that, do we want to do that in this PR or a new subtask?

File: app/views/components/radios.scala.html [hunk 7]
```diff
@@ -31,18 +31,9 @@
 )(implicit messages: Messages, userRequest: UserRequest[_], appConfig: AppConfig)
 
 @legend = {
```
Comment (LewisMitchell) [inline app/views/components/radios.scala.html:33]: This isn't being used, according to the HMRC design system - there should be a `<h1>` inside the `<legend>` I suspect it is to do with the check on line 48.

Comment (LewisMitchell) [reply app/views/components/radios.scala.html:33]: image

Comment (LewisMitchell) [reply app/views/components/radios.scala.html:33]: Be careful though as not all pages with fieldsets have legends which contain `<h1>` within them. For example the `/multiple-penalties-for-this-period` page has a h1 that isn't in the fieldset

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Comment by LewisMitchell** [comment]
Will need to rebase also due to #426 going in

File: app/views/components/radios.scala.html [hunk 8]
```diff
@@ -31,18 +31,9 @@
...
-      }
-    </span>
-    <span class="govuk-heading-l govuk-!-margin-bottom-0" id="page-heading">@legendContent</span>
-  </h1>
+    <h1 class="govuk-fieldset__heading" id ="page-heading">
+        @legendContent
+    </h1>
 }
 
 @govukRadios(Radios(
```
Comment (LewisMitchell) [inline app/views/components/radios.scala.html:39]: Could you rename showPenaltyInformation to be something like legendAsPageHeading?

File: app/views/components/inputDate.scala.html [hunk 9]
```diff
@@ -28,17 +28,8 @@
         legendAsPageHeading: Boolean = false)(implicit messages: Messages, userRequest: UserRequest[_], appConfig: AppConfig)
```
Comment (LewisMitchell) [inline app/views/components/inputDate.scala.html:28]: Do we need showPenaltyInformation now? It seems redundant as we never show penalty information within this component now

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/views/components/inputDate.scala.html [hunk 10]
```diff
@@ -24,21 +24,11 @@
         legendClasses: Option[String] = None,
         id: String = "date",
         hintText: Option[String] = None,
-        showPenaltyInformation: Boolean = true,
-        legendAsPageHeading: Boolean = false)(implicit messages: Messages, userRequest: UserRequest[_], appConfig: AppConfig)
+        showPenaltyInformation: Boolean = true)(implicit messages: Messages, userRequest: UserRequest[_], appConfig: AppConfig)
```
Comment (LewisMitchell) [inline app/views/components/inputDate.scala.html:27]: Should this be legendAsPageHeading?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**